sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3170,7 +3170,9 @@ class Analyzer(override val catalogManager: CatalogManager)
         WindowExpression(wf, s.copy(frameSpecification = wf.frame))
       case we @ WindowExpression(e, s @ WindowSpecDefinition(_, o, UnspecifiedFrame))
           if e.resolved =>
-        val frame = if (o.nonEmpty) {
+        val frame = if (e.isInstanceOf[RankLike]) {
+          SpecifiedWindowFrame(RowFrame, UnboundedPreceding, CurrentRow)
```

Comment:
what happens if order spec is present? do we force to use row frame?

Reply:
> what happens if order spec is present? do we force to use row frame?

In PostgresSQL, for all AggregationWindowFunction, it has a default frame, but user can write all kind of frame and finally it just use default frame. So in current PR, We can replace user-defined frame to default frame in `ResolveWindowFrame`, how about current change.

Reply:
It seems we should investigate the behavior of other database.

Reply:
Hive 2.3.7 don't allow to specify frame.
![WechatIMG75](https://user-images.githubusercontent.com/46485123/106589041-7fdac980-6586-11eb-885e-894a5431e344.jpeg)